torch/utils/data/_utils/pin_memory.py

```diff
@@ -15,7 +15,7 @@
 from . import MP_STATUS_CHECK_INTERVAL
 
 
-def _pin_memory_loop(in_queue, out_queue, device_id, done_event, device) -> None:
+def _pin_memory_loop(in_queue, out_queue, device_id, done_event):
```

Comment:
Even though these are "internal" APIs, it seems they're used by torchdata: https://github.com/meta-pytorch/data/blob/92950795e0790eb74df995daf40b658e85fd2c9f/torchdata/stateful_dataloader/stateful_dataloader.py#L1005-L1011

This change breaks StatefulDataLoader until the corresponding `_pin_memory_loop` call can be updated. I'll need to test it, but I think we might be able to just add a deprecation warning here in the meantime: 

```suggestion
def _pin_memory_loop(in_queue, out_queue, device_id, done_event, device=None):

    if device is not None:
        warnings.warn(
            "The device argument to _pin_memory_loop has been deprecated and will be removed in a "
            "future version of PyTorch."
            DeprecationWarning,
            stacklevel=2,
        )
```

Reply:
Done